packages/client/src/features/smart-connector/smart-connector.ts

```diff
@@ -0,0 +1,524 @@
+            }
+        }
+        this.hideSmartConnector();
+    }
+
+    protected async requestAvailableOptions(contextElement?: GModelElement): Promise<void> {
+        const requestAction = RequestContextActions.create({
+            contextId: SmartConnector.ID,
+            editorContext: {
+                selectedElementIds: [this.selectedElementId],
+                args: { nodeType: contextElement!.type }
+            }
+        });
+        const response = await this.actionDispatcher.request<SetContextActions>(requestAction);
+        this.smartConnectorItems = response.actions.map(e => e as SmartConnectorGroupItem);
+        this.createBody();
+    }
+
+    protected setMainPosition(viewport: ViewportResult, nodeBounds: DOMRect): void {
+        const xCenter = nodeBounds.x + nodeBounds.width/2 - viewport.canvasBounds.x;
+        const yCenter = nodeBounds.y + nodeBounds.height/2 - viewport.canvasBounds.y;
+        this.containerElement.style.left = `${xCenter}px`;
+        this.containerElement.style.top = `${yCenter}px`;
+    }
+
+    protected setPosition(element: HTMLElement, position: SmartConnectorPosition, nodeBounds: DOMRect, single?: boolean): void {
+        const zoom = this.currentZoom;
+        element.style.transform = `scale(${zoom})`;
+        const nodeHeight = nodeBounds.height;
+        const nodeWidth = nodeBounds.width;
+        let xDiff = -element.offsetWidth/2;
+        let yDiff = -element.offsetHeight/2*zoom;
+        if (position === SmartConnectorPosition.Right || position === SmartConnectorPosition.Left) {
+            xDiff = nodeWidth/2 + CONTAINER_PADDING_PX*zoom;
+            element.style.top = `${yDiff}px`;
+        }
+        if (position === SmartConnectorPosition.Top || position === SmartConnectorPosition.Bottom) {
+            yDiff = nodeHeight/2 + CONTAINER_PADDING_PX*zoom;
+            element.style.left = `${xDiff}px`;
+        }
+        if (position === SmartConnectorPosition.Right) {
+            element.style.transformOrigin = 'top left';
+            element.style.left = `${xDiff}px`;
+        }
+        if (position === SmartConnectorPosition.Left) {
+            element.style.transformOrigin = 'top right';
+            element.style.right = `${xDiff}px`;
+        }
+        if (position === SmartConnectorPosition.Top) {
+            element.style.transformOrigin = 'bottom';
+            element.style.bottom = `${yDiff}px`;
+        }
+        if (position === SmartConnectorPosition.Bottom) {
+            element.style.transformOrigin = 'top';
+            element.style.top = `${yDiff}px`;
+        }
+        if (single) {element.style.position = 'absolute';}
+    }
+
+    protected initializeContents(containerElement: HTMLElement): void {
+        this.createBody();
+        this.createExpandButton();
+        this.containerElement.appendChild(this.expandButton);
+        containerElement.setAttribute('aria-label', 'Smart-Connector');
+    }
+
+    protected createBody(): void {
```

Comment:
```suggestion
    protected initBody(): void {
```
you are not creating a new element. You are initializing the body. Create would mean, that you are returning something.